test/catch2/misc_tests/test_page_log_handle.cpp

```diff
@@ -0,0 +1,122 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "wiredtiger.h"
+#include "../wrappers/mock_session.h"
+
+int
+mock_plh_open_handle(
+  WT_PAGE_LOG *page_log, WT_SESSION *session, uint64_t table_id, WT_PAGE_LOG_HANDLE **plh)
+{
+    WT_PAGE_LOG_HANDLE *mock_log_handle;
+    WT_UNUSED(page_log);
+    WT_UNUSED(table_id);
+
+    WT_RET(__wt_calloc_one((WT_SESSION_IMPL *)session, &mock_log_handle));
+    *plh = mock_log_handle;
+    return (0);
+}
+
+int
+mock_terminate(WT_PAGE_LOG *page_log, WT_SESSION *session)
+{
+    __wt_free((WT_SESSION_IMPL *)session, page_log);
+    return (0);
+}
+
+int
+mock_plh_close(WT_PAGE_LOG_HANDLE *plh, WT_SESSION *session)
+{
+    WT_UNUSED(plh);
+    WT_UNUSED(session);
+    return (0);
+}
+
+void
+setup_page_log_queue(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn_impl = S2C(session);
+    WT_NAMED_PAGE_LOG *npl;
+
+    /* Create a mock page log. */
+    WT_PAGE_LOG *page_log;
+    REQUIRE(__wt_calloc_one(session, &page_log) == 0);
+    page_log->pl_open_handle = mock_plh_open_handle;
+    page_log->terminate = mock_terminate;
+
+    /* Create a named page log. */
+    REQUIRE(__wt_calloc_one(session, &npl) == 0);
+    npl->page_log = page_log;
+    REQUIRE(__wt_strdup(session, "mock", &npl->name) == 0);
+
+    /* Insert mock page log into mock connection. */
+    TAILQ_INIT(&conn_impl->pagelogqh);
+    TAILQ_INSERT_TAIL(&conn_impl->pagelogqh, npl, q);
+}
+
+TEST_CASE("Test disaggregated configuration logic", "[disagg_config]")
+{
+    std::shared_ptr<mock_session> session_mock = mock_session::build_test_mock_session();
+
+    WT_CONNECTION_IMPL *conn_impl = session_mock->get_mock_connection()->get_wt_connection_impl();
+    WT_SESSION_IMPL *session = session_mock->get_wt_session_impl();
+
+    const char *disagg_cfg[] = {"disaggregated=(role=follower,page_log=mock)", NULL};
+
+    /* Setup the page log queue. */
+    setup_page_log_queue(session);
+
+    SECTION("Test page log handle is constructed")
+    {
+        REQUIRE(__wti_disagg_conn_config(session, disagg_cfg, false) == WT_ERROR);
```

Comment:
Calling the function `__wti_disagg_conn_config` errors out after the page_log. There are problems setting up disaggregate via unit-test. I am planning to create another ticket to tackle this on a high level.

For the sake of this test, the page log handle is properly constructed with the mock page log. Thus there is value in this unit test.